File: file/zq200.js
// ============================================================
// zuqiu200直播爬虫 v1 (CatVod格式)
// 将Python版本转换为TVBox兼容的JS格式
// ============================================================

const API_HOST = 'https://enskweeseey8kp2frvb06.k8v4dh4.app';
const HOST = 'https://zuqiu200.app';
const UA = 'Mozilla/5.0 (Windows NT 10.0; Win64; x64) AppleWebKit/537.36';

// ========== 工具函数 ==========
function getHeaders() {
    return {
        'User-Agent': UA,
        'Referer': HOST + '/',
        'Origin': HOST,
        'Accept': 'application/json, text/plain, */*',
        'Accept-Language': 'zh-CN,zh;q=0.9,en;q=0.8'
    };
}

async function fetchJson(url) {
    try {
        const hd = getHeaders();
        if (typeof Java !== 'undefined' && Java && Java.req) {
            const r = await Java.req(url, { headers: hd });
            if (typeof r === 'string') return JSON.parse(r);
            const code = Number((r && (r.statusCode || r.status || r.code)) || 0);
            const body = String((r && (r.body || r.content || r.data)) || '');
            if (code >= 200 && code < 300 && body) return JSON.parse(body);
            return {};
        }
        const r2 = await req(url, { headers: hd });
        if (typeof r2 === 'string') return JSON.parse(r2);
        const code2 = Number((r2 && (r2.statusCode || r2.status || r2.code)) || 0);
        const body2 = String((r2 && (r2.content || r2.body || r2.data)) || '');
        if (code2 >= 200 && code2 < 300 && body2) return JSON.parse(body2);
        return {};
    } catch (e) {
        return {};
    }
}

function fmtTime(tsSec) {
    if (!tsSec) return '';
    try {
        const d = new Date(tsSec * 1000);
        const pad = n => String(n).padStart(2, '0');
        return pad(d.getMonth() + 1) + '-' + pad(d.getDate()) + ' ' + pad(d.getHours()) + ':' + pad(d.getMinutes());
    } catch (e) {
        return '';
    }
}

function getLogoUrl(sportId, logo) {
    if (!logo) return '';
    if (logo.startsWith('http')) return logo;
    if (logo.includes('/')) return 'https://img0.aiscore.com/' + logo;
    const prefix = {1: 'football', 2: 'basketball', 3: 'v2', 4: 'volleyball'};
    return 'https://img0.aiscore.com/' + (prefix[sportId] || 'football') + '/team/' + logo;
}

// ========== 分类 ==========
function getClasses() {
    return [
        { type_id: '1', type_name: '足球' },
        { type_id: '2', type_name: '篮球' },
        { type_id: '3', type_name: '网球' },
        { type_id: '4', type_name: '排球' },
        { type_id: 'all', type_name: '全部' }
    ];
}

// ========== 获取直播数据 ==========
async function getLivestreams() {
    const ts = Date.now();
    const url = API_HOST + '/api/c5/business/livehouse/index?lang=zh&timestamp=' + ts;
    const json = await fetchJson(url);
    if (!json || !json.success) return [];
    
    const result = [];
    const seen = new Set();
    const sportPrefix = {1: 'football', 2: 'basketball', 3: 'v2', 4: 'volleyball'};
    
    const matchStreams = (json.data && json.data.matchLivestreams) || [];
    const houseTimeMap = {};
    const houseStreamMap = {};
    
    // 构建houseId到比赛时间和流的映射
    for (let i = 0; i < matchStreams.length; i++) {
        const item = matchStreams[i];
        const matchData = item.result && item.result.match;
        if (!matchData) continue;
        const matchTime = matchData.matchTime || 0;
        const statusId = matchData.statusId || 0;
        const anchors = item.reservedAnchors || [];
        for (let j = 0; j < anchors.length; j++) {
            const hId = anchors[j].houseId;
            if (hId) houseTimeMap[hId] = [matchTime, statusId];
        }
    }
    
    // 处理正在进行的直播流
    const ongoing = (json.data && json.data.ongoingLivestreams) || [];
    for (let i = 0; i < ongoing.length; i++) {
        const item = ongoing[i];
        const houseId = item.houseId;
        const streamUrl = item.playStreamAddress;
        const streamM3u8 = item.playStreamAddress2;
        let validStream = '';
        if (streamUrl && streamUrl.startsWith('http') && streamUrl !== 'www') validStream = streamUrl;
        else if (streamM3u8 && streamM3u8.startsWith('http') && streamM3u8 !== 'www') validStream = streamM3u8;
        if (houseId && validStream) houseStreamMap[houseId] = validStream;
    }
    
    // 处理主播直播流
    const anchorStreams = (json.data && json.data.anchorLivestreams) || [];
    for (let i = 0; i < anchorStreams.length; i++) {
        const item = anchorStreams[i];
        const houseId = item.houseId;
        const streamUrl = item.playStreamAddress;
        const streamM3u8 = item.playStreamAddress2;
        let validStream = '';
        if (streamUrl && streamUrl.startsWith('http') && streamUrl !== 'www') validStream = streamUrl;
        else if (streamM3u8 && streamM3u8.startsWith('http') && streamM3u8 !== 'www') validStream = streamM3u8;
        if (houseId && validStream) houseStreamMap[houseId] = validStream;
    }
    
    // 处理比赛流
    for (let i = 0; i < matchStreams.length; i++) {
        const item = matchStreams[i];
        const matchData = item.result && item.result.match;
        if (!matchData) continue;
        const competition = matchData.competition || {};
        const homeTeamData = matchData.homeTeam || {};
        const awayTeamData = matchData.awayTeam || {};
        const homeTeam = homeTeamData.name || '';
        const awayTeam = awayTeamData.name || '';
        const league = competition.name || '';
        const videoUrl = matchData.videoUrl || '';
        const sportId = matchData.sportId;
        const statusId = matchData.statusId || 0;
        const matchName = homeTeam && awayTeam ? homeTeam + ' vs ' + awayTeam : '';
        if (!matchName) continue;
        
        let effectiveVideoUrl = '';
        if (videoUrl && videoUrl.startsWith('http') && videoUrl !== 'https') {
            effectiveVideoUrl = videoUrl;
        } else {
            const anchors = item.reservedAnchors || [];
            for (let j = 0; j < anchors.length; j++) {
                const anchorStream = anchors[j].playStreamAddress || '';
                const anchorStream2 = anchors[j].playStreamAddress2 || '';
                if (anchorStream && anchorStream.startsWith('http') && anchorStream !== 'www') {
                    effectiveVideoUrl = anchorStream;
                    break;
                } else if (anchorStream2 && anchorStream2.startsWith('http') && anchorStream2 !== 'www') {
                    effectiveVideoUrl = anchorStream2;
                    break;
                }
            }
            if (!effectiveVideoUrl) {
                for (let j = 0; j < anchors.length; j++) {
                    const hId = anchors[j].houseId;
                    if (hId && houseStreamMap[hId]) {
                        effectiveVideoUrl = houseStreamMap[hId];
                        break;
                    }
                }
            }
        }
        
        const matchId = matchData.id || '';
        const matchTime = matchData.matchTime || 0;
        const timeStr = fmtTime(matchTime);
        const isLive = effectiveVideoUrl && (statusId === 2 || statusId === 4);
        const isUpcoming = !effectiveVideoUrl && (statusId === 1 || statusId === 2 || statusId === 3);
        const prefix = isLive ? '🟢 ' : isUpcoming ? '⏳ ' : '';
        const remarks = timeStr && league ? prefix + timeStr + ' ' + league : timeStr ? prefix + timeStr : league ? prefix + league : '';
        const homeLogo = homeTeamData.logo || '';
        const homeLogoUrl = getLogoUrl(sportId, homeLogo);
        const key = matchName + '_' + league;
        if (seen.has(key)) continue;
        seen.add(key);
        const finalUrl = effectiveVideoUrl ? effectiveVideoUrl : (matchId ? 'https://zuqiu200.app/match/' + matchId : '');
        
        result.push({
            vod_id: matchId,
            vod_name: matchName,
            vod_pic: homeLogoUrl,
            vod_play_from: league || '直播',
            vod_play_url: finalUrl,
            vod_remarks: remarks,
            sport_id: String(sportId || '')
        });
    }
    
    // 处理进行中的直播
    for (let i = 0; i < ongoing.length; i++) {
        const item = ongoing[i];
        const houseId = item.houseId;
        const houseName = item.houseName || '';
        const streamUrl = item.playStreamAddress;
        const streamM3u8 = item.playStreamAddress2;
        const raceTypeId = item.raceTypeId;
        const houseImage = item.houseImage || '';
        
        let validStream = '';
        if (streamUrl && streamUrl.startsWith('http') && streamUrl !== 'www') validStream = streamUrl;
        else if (streamM3u8 && streamM3u8.startsWith('http') && streamM3u8 !== 'www') validStream = streamM3u8;
        
        if (houseId && houseName && validStream) {
            const parts = houseName.split('|');
            const league = parts.length > 0 ? parts[0].trim() : '';
            const matchInfo = parts.length > 1 ? parts[1].trim() : houseName;
            let teamParts = matchInfo.split('VS');
            if (teamParts.length < 2) teamParts = matchInfo.split('vs');
            const homeTeam = teamParts.length > 0 ? teamParts[0].trim() : '';
            const awayTeam = teamParts.length > 1 ? teamParts[1].trim() : '';
            const matchName = homeTeam && awayTeam ? homeTeam + ' vs ' + awayTeam : houseName;
            const mt = houseTimeMap[houseId] || [0, 0];
            const timeStr = fmtTime(mt[0]);
            const remarks = timeStr && league ? '🟢 ' + timeStr + ' ' + league : timeStr ? '🟢 ' + timeStr : league ? '🟢 ' + league : '🟢 直播中';
            const key = matchName + '_' + league;
            if (seen.has(key)) continue;
            seen.add(key);
            
            result.push({
                vod_id: houseId,
                vod_name: matchName,
                vod_pic: houseImage,
                vod_play_from: league || '直播',
                vod_play_url: validStream,
                vod_remarks: remarks,
                sport_id: String(raceTypeId || '')
            });
        }
    }
    
    return result;
}

// ========== 公开接口 ==========
async function init(cfg) {}

async function home(filter) {
    return JSON.stringify({
        class: getClasses(),
        filters: {}
    });
}

async function homeVod() {
    return await category('all', 1, false, {});
}

async function category(tid, pg, filter, extend) {
    tid = String((extend && extend.cateId) || tid || 'all');
    pg = parseInt(pg) || 1;
    let list = [];
    
    try {
        const allVideos = await getLivestreams();
        if (tid === 'all') {
            list = allVideos;
        } else {
            list = allVideos.filter(v => v.sport_id === tid);
        }
        
        const pageSize = 20;
        const start = (pg - 1) * pageSize;
        const end = start + pageSize;
        list = list.slice(start, end);
    } catch (e) {
        list = [];
    }
    
    return JSON.stringify({
        code: 1,
        msg: '数据列表',
        page: pg,
        pagecount: Math.ceil(list.length / 20) || 1,
        limit: 20,
        total: list.length,
        list: list
    });
}

async function detail(ids) {
    const id = String(Array.isArray(ids) ? ids[0] : ids || '');
    if (!id) return JSON.stringify({ code: 1, list: [] });
    
    try {
        const allVideos = await getLivestreams();
        for (let i = 0; i < allVideos.length; i++) {
            if (allVideos[i].vod_id === id) {
                const detail = Object.assign({}, allVideos[i]);
                return JSON.stringify({
                    code: 1,
                    list: [detail]
                });
            }
        }
    } catch (e) {}
    
    return JSON.stringify({ code: 1, list: [] });
}

async function search(wd, quick, pg) {
    pg = parseInt(pg) || 1;
    let results = [];
    
    try {
        const allVideos = await getLivestreams();
        const keywordLower = wd.toLowerCase();
        
        for (let i = 0; i < allVideos.length; i++) {
            const video = allVideos[i];
            const name = video.vod_name.toLowerCase();
            if (name.includes(keywordLower) || video.vod_play_from.includes(wd)) {
                results.push(video);
            }
        }
        
        const pageSize = 20;
        const start = (pg - 1) * pageSize;
        const end = start + pageSize;
        results = results.slice(start, end);
    } catch (e) {
        results = [];
    }
    
    return JSON.stringify({
        code: 1,
        msg: '搜索结果',
        page: pg,
        pagecount: Math.ceil(results.length / 20) || 1,
        limit: 20,
        total: results.length,
        list: results
    });
}

async function play(flag, id, flags) {
    const url = String(id || '');
    if (!url) {
        return JSON.stringify({
            parse: 0,
            url: '',
            header: ''
        });
    }
    
    return JSON.stringify({
        parse: 0,
        url: url,
        header: {
            'User-Agent': UA,
            'Referer': HOST + '/',
            'Origin': HOST
        }
    });
}

// ========== 导出接口 ==========
async function homeContent(filter) {
    try {
        return JSON.parse(await home(filter));
    } catch (e) {
        return { class: [], list: [], filter: {} };
    }
}

async function homeVideoContent() {
    try {
        return JSON.parse(await category('all', 1, false, {}));
    } catch (e) {
        return { list: [] };
    }
}

async function categoryContent(tid, pg, filter, extend) {
    try {
        return JSON.parse(await category(tid, pg, filter, extend || {}));
    } catch (e) {
        return { list: [] };
    }
}

async function detailContent(ids) {
    try {
        return JSON.parse(await detail(ids));
    } catch (e) {
        return { list: [] };
    }
}

async function searchContent(wd, quick, pg) {
    try {
        return JSON.parse(await search(wd, quick, pg || 1));
    } catch (e) {
        return { list: [] };
    }
}

async function playerContent(flag, id, flags) {
    try {
        return JSON.parse(await play(flag, id, flags));
    } catch (e) {
        return { parse: 0, url: id || '', header: '' };
    }
}

export function __jsEvalReturn() {
    return {
        init,
        home,
        homeVod,
        category,
        search,
        detail,
        play,
        homeContent,
        homeVideoContent,
        categoryContent,
        detailContent,
        searchContent,
        playerContent
    };
}

File: file/zq200.py
#!/usr/bin/python
# -*- coding: utf-8 -*-
# Version: V12 - 无外部依赖版本，使用urllib替代requests
# 适用于  等平台，无需pip安装任何依赖,过滤主播
import json, time, re
from urllib.request import urlopen, Request

try:
    from base.spider import Spider
except ImportError:
    # 独立运行时的备用Spider类
    class Spider:
        def getName(self): return ""
        def init(self, extend=""): pass

class Spider(Spider):
    def getName(self): return "zuqiu200直播"
    def init(self, extend=""):
        self.host = "https://zuqiu200.app"
        self.api_host = "https://enskweeseey8kp2frvb06.k8v4dh4.app"
        self.headers = {"User-Agent": "Mozilla/5.0 (Windows NT 10.0; Win64; x64) AppleWebKit/537.36", "Referer": self.host + "/", "Origin": self.host}
        self.categories = [
            {"type_id": "1", "type_name": "足球"},
            {"type_id": "2", "type_name": "篮球"},
            {"type_id": "3", "type_name": "网球"},
            {"type_id": "4", "type_name": "排球"},
            {"type_id": "all", "type_name": "全部"}
        ]
    
    def _api_get(self, path):
        url = f"{self.api_host}{path}"
        try:
            req = Request(url, headers=self.headers)
            with urlopen(req, timeout=15) as resp:
                if resp.getcode() == 200:
                    data = resp.read().decode()
                    return json.loads(data)
            return None
        except: return None
    
    def _get_livestreams(self):
        ts = int(time.time() * 1000)
        data = self._api_get(f"/api/c5/business/livehouse/index?lang=zh&timestamp={ts}")
        if not data or not data.get("success"): return []
        result = []
        seen = set()
        sport_prefix = {1: "football", 2: "basketball", 3: "v2", 4: "volleyball"}
        
        def get_logo_url(sport_id, logo):
            if logo:
                if not logo.startswith("http"):
                    if "/" in logo:
                        return f"https://img0.aiscore.com/{logo}"
                    else:
                        prefix = sport_prefix.get(sport_id, "football")
                        return f"https://img0.aiscore.com/{prefix}/team/{logo}"
            return logo or ""
        
        def fmt_time(ts_sec):
            if not ts_sec: return ""
            return time.strftime('%m-%d %H:%M', time.localtime(ts_sec))
        
        match_streams = data.get("data", {}).get("matchLivestreams", [])
        house_time_map = {}
        house_stream_map = {}
        for item in match_streams:
            match_data = item.get("result", {}).get("match", {})
            if not match_data: continue
            match_time = match_data.get("matchTime", 0)
            status_id = match_data.get("statusId", 0)
            for anchor in item.get("reservedAnchors", []):
                h_id = anchor.get("houseId", "")
                if h_id: house_time_map[h_id] = (match_time, status_id)
        
        ongoing = data.get("data", {}).get("ongoingLivestreams", [])
        for item in ongoing:
            house_id = item.get("houseId", "")
            stream_url = item.get("playStreamAddress", "")
            stream_m3u8 = item.get("playStreamAddress2", "")
            valid_stream = ""
            if stream_url and stream_url.startswith("http") and stream_url != "www":
                valid_stream = stream_url
            elif stream_m3u8 and stream_m3u8.startswith("http") and stream_m3u8 != "www":
                valid_stream = stream_m3u8
            if house_id and valid_stream:
                house_stream_map[house_id] = valid_stream
        
        anchor_streams = data.get("data", {}).get("anchorLivestreams", [])
        for item in anchor_streams:
            house_id = item.get("houseId", "")
            stream_url = item.get("playStreamAddress", "")
            stream_m3u8 = item.get("playStreamAddress2", "")
            valid_stream = ""
            if stream_url and stream_url.startswith("http") and stream_url != "www":
                valid_stream = stream_url
            elif stream_m3u8 and stream_m3u8.startswith("http") and stream_m3u8 != "www":
                valid_stream = stream_m3u8
            if house_id and valid_stream:
                house_stream_map[house_id] = valid_stream
        
        for item in match_streams:
            match_data = item.get("result", {}).get("match", {})
            if not match_data: continue
            competition = match_data.get("competition", {})
            home_team_data = match_data.get("homeTeam", {})
            away_team_data = match_data.get("awayTeam", {})
            home_team = home_team_data.get("name", "")
            away_team = away_team_data.get("name", "")
            league = competition.get("name", "")
            video_url = match_data.get("videoUrl", "")
            sport_id = match_data.get("sportId")
            status_id = match_data.get("statusId", 0)
            match_name = f"{home_team} vs {away_team}" if home_team and away_team else ""
            if not match_name: continue
            
            effective_video_url = ""
            if video_url and video_url.startswith("http") and video_url != "https":
                effective_video_url = video_url
            else:
                for anchor in item.get("reservedAnchors", []):
                    anchor_stream = anchor.get("playStreamAddress", "")
                    anchor_stream2 = anchor.get("playStreamAddress2", "")
                    if anchor_stream and anchor_stream.startswith("http") and anchor_stream != "www":
                        effective_video_url = anchor_stream
                        break
                    elif anchor_stream2 and anchor_stream2.startswith("http") and anchor_stream2 != "www":
                        effective_video_url = anchor_stream2
                        break
                
                if not effective_video_url:
                    for anchor in item.get("reservedAnchors", []):
                        h_id = anchor.get("houseId", "")
                        if h_id and h_id in house_stream_map:
                            effective_video_url = house_stream_map[h_id]
                            break
            
            match_id = match_data.get("id", "")
            match_time = match_data.get("matchTime", 0)
            time_str = fmt_time(match_time)
            is_live = effective_video_url and status_id in (2, 4)
            is_upcoming = not effective_video_url and status_id in (1, 2, 3)
            if is_live:
                prefix = "🟢 "
            elif is_upcoming:
                prefix = "⏳ "
            else:
                prefix = ""
            if time_str and league:
                remarks = f"{prefix}{time_str} {league}"
            elif time_str:
                remarks = f"{prefix}{time_str}"
            else:
                remarks = f"{prefix}{league}" if league else ""
            home_logo = home_team_data.get("logo", "")
            home_logo_url = get_logo_url(sport_id, home_logo)
            key = f"{match_name}_{league}"
            if key in seen: continue
            seen.add(key)
            if effective_video_url:
                final_url = effective_video_url
            else:
                final_url = f"https://zuqiu200.app/match/{match_id}" if match_id else ""
            result.append({
                "vod_id": match_id,
                "vod_name": match_name,
                "vod_pic": home_logo_url,
                "vod_play_from": league or "直播",
                "vod_play_url": final_url,
                "vod_remarks": remarks,
                "sport_id": str(sport_id) if sport_id else ""
            })
        
        for item in ongoing:
            house_id = item.get("houseId", "")
            house_name = item.get("houseName", "")
            stream_url = item.get("playStreamAddress", "")
            stream_m3u8 = item.get("playStreamAddress2", "")
            race_type_id = item.get("raceTypeId")
            house_image = item.get("houseImage", "")
            
            valid_stream = ""
            if stream_url and stream_url.startswith("http") and stream_url != "www":
                valid_stream = stream_url
            elif stream_m3u8 and stream_m3u8.startswith("http") and stream_m3u8 != "www":
                valid_stream = stream_m3u8
            
            if house_id and house_name and valid_stream:
                parts = house_name.split("|")
                league = parts[0].strip() if len(parts) > 0 else ""
                match_info = parts[1].strip() if len(parts) > 1 else house_name
                team_parts = match_info.split("VS")
                if len(team_parts) < 2: team_parts = match_info.split("vs")
                home_team = team_parts[0].strip() if len(team_parts) > 0 else ""
                away_team = team_parts[1].strip() if len(team_parts) > 1 else ""
                match_name = f"{home_team} vs {away_team}" if home_team and away_team else house_name
                mt = house_time_map.get(house_id, (0, 0))
                time_str = fmt_time(mt[0])
                if time_str and league:
                    remarks = f"🟢 {time_str} {league}"
                elif time_str:
                    remarks = f"🟢 {time_str}"
                else:
                    remarks = f"🟢 {league}" if league else "🟢 直播中"
                key = f"{match_name}_{league}"
                if key in seen: continue
                seen.add(key)
                result.append({
                    "vod_id": house_id,
                    "vod_name": match_name,
                    "vod_pic": house_image,
                    "vod_play_from": league or "直播",
                    "vod_play_url": valid_stream,
                    "vod_remarks": remarks,
                    "sport_id": str(race_type_id) if race_type_id else ""
                })
        
        return result
    
    def homeVideoContent(self, extend=""):
        return self.homeContent(extend)
    
    def homeContent(self, extend=""):
        try:
            classes = []
            for cat in self.categories:
                classes.append({
                    "type_id": cat["type_id"],
                    "type_name": cat["type_name"],
                    "type_extend": "",
                    "type_url": ""
                })
            
            videos = self._get_livestreams()
            return {"class": classes, "list": videos, "filter": {}}
        except Exception as e:
            return {"class": [], "list": [], "filter": {}}
    
    def categoryContent(self, tid, pg, filterable, extend):
        try:
            all_videos = self._get_livestreams()
            if tid == "all":
                videos = all_videos
            else:
                videos = [v for v in all_videos if v.get("sport_id") == tid]
            
            page_size = 20
            start = (int(pg) - 1) * page_size
            end = start + page_size
            page_videos = videos[start:end]
            
            return {"list": page_videos, "page": pg, "pagecount": (len(videos) + page_size - 1) // page_size, "limit": page_size, "total": len(videos)}
        except Exception as e:
            return {"list": [], "page": pg, "pagecount": 1, "limit": 20, "total": 0}
    
    def detailContent(self, ids):
        try:
            all_videos = self._get_livestreams()
            for video in all_videos:
                if video["vod_id"] == ids[0]:
                    detail = video.copy()
                    detail["vod_play_from"] = video["vod_play_from"]
                    detail["vod_play_url"] = video["vod_play_url"]
                    return {"list": [detail]}
            return {"list": []}
        except Exception as e:
            return {"list": []}
    
    def searchContent(self, keyword, quick, pg):
        try:
            all_videos = self._get_livestreams()
            results = []
            keyword_lower = keyword.lower()
            
            for video in all_videos:
                name = video["vod_name"].lower()
                if keyword_lower in name or keyword in video["vod_play_from"]:
                    results.append(video)
            
            page_size = 20
            start = (int(pg) - 1) * page_size
            end = start + page_size
            page_results = results[start:end]
            
            return {"list": page_results, "page": pg, "pagecount": (len(results) + page_size - 1) // page_size, "limit": page_size, "total": len(results)}
        except Exception as e:
            return {"list": [], "page": pg, "pagecount": 1, "limit": 20, "total": 0}
    
    def playerContent(self, flag, id, vipFlags):
        try:
            parse = 0
            url = id
            
            if not url:
                url = ""
            
            return {"parse": parse, "url": url, "header": ""}
        except Exception as e:
            return {"parse": 0, "url": "", "header": ""}